infra/src/test/kotlin/fi/fta/geoviite/infra/linking/switches/SwitchFittingTest.kt

```diff
@@ -31,12 +43,22 @@ data class TrackForSwitchFitting(
         return geometry.getPointAtM(m)
     }
 
+    fun setTrackNumber(trackNumberId: IntId<LayoutTrackNumber>): TrackForSwitchFitting {
+        return TrackForSwitchFitting(byJoints, locationTrack.copy(trackNumberId = trackNumberId), geometry)
+    }
+
     fun cutFromStart(length: Double): TrackForSwitchFitting {
         return cutFromStart(locationTrack, geometry, length).let { (locationTrack, geometry) ->
             this.copy(locationTrack = locationTrack, geometry = geometry)
```

Comment:
this:iä ei pitäisi tarvita näissä

Reply:
Mitenkäs noille muille val-määrityksille muuten käy jos tekee kopion näin? Evaluoikos se nuo uudelleen? Luulisin että kyllä muttan ole ihan varma - jos ei niin nuo voi toki vaihtaa muotoon `get() = ...`